Libraries/src/Amazon.Lambda.ApplicationLoadBalancerIdentity/ALBIdentityMiddleware.cs

```diff
@@ -0,0 +1,245 @@
+                else
+                    this.cacheDuration = maxCacheLife;
+
+                // Populate the byte lookup array.
+                this.sha = SHA1.Create();
+                byteLookup = new uint[256];
+                for (int i = 0; i < 256; i++)
+                {
+                    var s = i.ToString("X2");
+                    byteLookup[i] = s[0] + ((uint)s[1] << 16);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method to invoke the middleware. Called by the system.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> of the current request.</param>
+        public async Task Invoke(HttpContext httpContext)
+        {
+            if (!httpContext.Request.Headers.TryGetValue(OidcDataHeader, out StringValues oidcData))
+            {
+                await BadRequest(httpContext, $"Missing header '{OidcDataHeader}'");
+                return;
+            }
+
+            if (this.cacheIdentities)
+            {
+                // We're going to key the identities based on the SHA1 hash of the payload.
+                // It would also be possible to parse the token and use the "Id" property,
+                // but performance tests showed a huge difference (188 ticks for hash vs 65283 for JWT parsing).
+                var hash = ByteArrayToHex(this.sha.ComputeHash(Encoding.UTF8.GetBytes(oidcData)));
+
+                try
+                {
+                    httpContext.User = await this.cachedIds.GetOrCreateAsync(hash, (entry) => this.GetUser(new JsonWebToken(oidcData), entry));
+                }
+                catch (Exception ex)
+                {
+                    this.cachedIds.Remove(hash);
+                    this.logger?.LogError($"{ex}");
+                    await BadRequest(httpContext, $"Invalid data in header '{OidcDataHeader}'");
+                    return;
+                }
+            }
+            else
+            {
+                try
+                {
+                    httpContext.User = await this.GetUser(new JsonWebToken(oidcData), null);
+                }
+                catch (Exception ex)
+                {
+                    this.logger?.LogError($"{ex}");
+                    await BadRequest(httpContext, $"Invalid data in header '{OidcDataHeader}': {ex.Message}");
+                    return;
+                }
+            }
+
+            // Invoke the next delegate in the pipeline
+            await this.next.Invoke(httpContext);
+        }
+
+        private async Task<ClaimsPrincipal> GetUser(JsonWebToken jwt, ICacheEntry cacheEntry)
+        {
+            this.logger?.LogDebug("User Principal is '{0}' (issued by '{1}')", jwt.Subject, jwt.Issuer);
+
+            if (this.options.VerifyTokenSignature)
+            {
+                var validationParameters = await this.validationParameters.GetOrAdd(jwt.Kid, async (key) =>
```

Comment:
Local variables and member variables having the same name cause too much confusion. Please use different names.

Reply:
Will do